upup/pkg/fi/cloudup/template_functions.go

```diff
@@ -128,11 +130,21 @@ func (tf *TemplateFunctions) GetInstanceGroup(name string) (*kops.InstanceGroup,
 	return nil, fmt.Errorf("InstanceGroup %q not found", name)
 }
 
+// DnsControllerArgv returns the args to the DNS controller
 func (tf *TemplateFunctions) DnsControllerArgv() ([]string, error) {
 	var argv []string
 
 	argv = append(argv, "/usr/bin/dns-controller")
 
+	if tf.cluster.Spec.DNSController != nil {
```

Comment:
FYI, I tend to do this:

```
dnsController := tf.cluster.Spec.DNSController
if dnsController == nil {
  dnsController = &DNSControllerSpec{}
}
...
```

With one option it's a wash, with a bunch it makes things simpler :-)